contracts/exchange/JitOrderHandler.sol

```diff
@@ -0,0 +1,153 @@
+            _orderVault,
+            _swapHandler,
+            _referralStorage
+        )
+    {
+        orderHandler = _orderHandler;
+        glvShiftHandler = _glvShiftHandler;
+    }
+
+    function executeJitOrder(
+        GlvShiftUtils.CreateGlvShiftParams[] memory shiftParamsList,
+        bytes32 orderKey,
+        OracleUtils.SetPricesParams calldata oracleParams
+    ) external override globalNonReentrant onlyOrderKeeper withOraclePrices(oracleParams) {
+        uint256 startingGas = gasleft();
+
+        DataStore _dataStore = dataStore;
+        FeatureUtils.validateFeature(_dataStore, Keys.jitFeatureDisabledKey(address(this)));
+
+        Order.Props memory order = OrderStoreUtils.get(_dataStore, orderKey);
+
+        _processShifts(_dataStore, shiftParamsList, order, orderKey);
+
+        uint256 estimatedGasLimit = GasUtils.estimateExecuteOrderGasLimit(dataStore, order);
+        // use gasleft() instead of startingGas to account for gas spent on GLV shifts
+        GasUtils.validateExecutionGas(dataStore, gasleft(), estimatedGasLimit);
+
+        orderHandler.executeOrderForController(
+            orderKey,
+            order,
+            startingGas,
+            GasUtils.getExecutionGas(_dataStore, gasleft()),
+            false // isSimulation
+        );
+    }
+
+    function simulateExecuteJitOrder(
+        GlvShiftUtils.CreateGlvShiftParams[] memory shiftParamsList,
+        bytes32 orderKey,
+        OracleUtils.SimulatePricesParams memory oracleParams
+    ) external globalNonReentrant withSimulatedOraclePrices(oracleParams) {
+        uint256 startingGas = gasleft();
+
+        DataStore _dataStore = dataStore;
+        FeatureUtils.validateFeature(_dataStore, Keys.jitFeatureDisabledKey(address(this)));
+        Order.Props memory order = OrderStoreUtils.get(_dataStore, orderKey);
+
+        _processShifts(_dataStore, shiftParamsList, order, orderKey);
+
+        orderHandler.executeOrderForController(
+            orderKey,
+            order,
+            startingGas,
+            GasUtils.getExecutionGas(_dataStore, gasleft()),
+            true // isSimulation
+        );
+    }
+
+    function _processShifts(
+        DataStore _dataStore,
+        GlvShiftUtils.CreateGlvShiftParams[] memory shiftParamsList,
+        Order.Props memory order,
+        bytes32 orderKey
+    ) internal {
+        if (shiftParamsList.length == 0) {
+            revert Errors.JitEmptyShiftParams();
+        }
+
+        for (uint256 i = 0; i < shiftParamsList.length; i++) {
+            if (order.market() != shiftParamsList[i].toMarket) {
+                revert Errors.JitInvalidToMarket(shiftParamsList[i].toMarket, order.market());
+            }
+
+            (bytes32 glvShiftKey, GlvShift.Props memory glvShift) = _createGlvShift(
+                _dataStore,
+                shiftParamsList[i],
+                order.updatedAtTime(),
+                orderKey
+            );
+
+            glvShiftHandler._executeGlvShift(
```

Comment:
guess it's better to rename `_executeGlvShift` or create new function, underscore usually means it's private/internal